storage/Utils/SystemCmd.cc

```diff
@@ -88,14 +88,23 @@ namespace storage
     void
     SystemCmd::init()
     {
+        _childStdin = NULL;
 	_files[0] = _files[1] = NULL;
-	_pfds[0].events = _pfds[1].events = POLLIN;
+	_pfds[0].events = POLLOUT; // stdin
+	_pfds[1].events = POLLIN;  // stdout
+	_pfds[2].events = POLLIN;  // stderr
```

Comment:
This is kind of tempting, but those defines don't exactly correspond. I ordered the file descriptors like that for convention, but as a matter of fact, in that _pfds array they could have any order as long as the application knows what they are all about. And yes, that whole thing is kind of hacky (it's late 80s style C to be exact).